explicit-bzero/explicit-bzero.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Portable Snippets - https://github.com/nemequ/portable-snippets
+ * Created by arnavyc <arnavyc@outlook.com>
+ *
+ *   To the extent possible under law, the authors have waived all
+ *   copyright and related or neighboring rights to this code.  For
+ *   details, see the Creative Commons Zero 1.0 Universal license at
+ *   https://creativecommons.org/publicdomain/zero/1.0/
+ *
+ */
+
+#ifndef __STDC_WANT_LIB_EXT1__
+/* memset_s() function in C11 Annex K (bound-checking interface) */
+#define __STDC_WANT_LIB_EXT1__ 1
+#endif
+
+#include <stdlib.h>
+#include <string.h>
+
+#include "explicit-bzero.h"
+
+#if defined(__FreeBSD__) || defined(__OpenBSD__)
+#include <sys/param.h>
+#endif
+
+#ifdef _WIN32
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+#endif /* _WIN32 */
+
+#ifdef __STDC_LIB_EXT1__
+/* memset_s() function in C11 Annex K (bound-checking interface) */
+#define HAVE_MEMSET_S 1
```

Comment:
I'd rather move these checks into the function.  We only need to check `HAVE_MEMSET_S` once, so IMHO it's better to keep the logic near the code.